LayoutTests/editing/selection/ios/navigating-in-subframe-does-not-dismiss-edit-menu.html

```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true ] -->
+<html>
+<head>
+<script src="../../../resources/js-test.js"></script>
+<script src="../../../resources/ui-helper.js"></script>
+<meta name=viewport content="width=device-width, initial-scale=1">
+<style>
+body, html {
+    width: 100%;
+    height: 100%;
+    margin: 0;
+}
+
+h1 {
+    font-size: 40px;
+}
+
+#target {
+    border: solid tomato 1px;
+    padding: 2px;
+}
+
+iframe {
+    width: 300px;
+    height: 100px;
+}
+</style>
+<script>
+jsTestIsAsync = true;
+subFrameLoaded = true;
+
+description("This test verifies that subframe navigation does not dismiss the edit menu. This test requires WebKitTestRunner.");
+
+addEventListener("load", async () => {
+    const subframe = document.querySelector("iframe");
+    subframe.addEventListener("load", () => { subFrameLoaded = true; });
+    subframe.srcdoc = "<p>subframe before reload</p>";
+    await shouldBecomeEqual("subFrameLoaded", "true");
+
+    if (!window.testRunner)
+        return;
+
+    const target = document.getElementById("target");
+    await UIHelper.activateElementAndWaitForInputSession(target);
+    document.execCommand("SelectAll", true);
+    testPassed("Selected text");
+
+    await UIHelper.activateElement(target);
+    await UIHelper.waitForMenuToShow();
+
+    subframe.srcdoc = `<p>subframe after reloading</p>`;
+    await UIHelper.delayFor(800);
```

Comment:
The 800ms delay doesn't have a specific meaning right? Is it possible to make it shorter and/or wait for something specific like a presentation update? 😬

Reply:
So this delay allows the edit menu to dismiss, in the case where the bug reproduces (and the editor state is erroneously cleared in the UI process as a result of subframe navigation). It's tricky to avoid _some_ hard-coded delay here, since the test needs to check that the edit menu _doesn't_ dismiss.